optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -54,14 +54,25 @@ def __init__(
 
     def compile(self):
         if self.request is None:
-            logger.info(f"Compiling the Language model to {self._device} ...")
-            self.request = core.compile_model(self.model, self._device, self.ov_config).create_infer_request()
+            if self._compile_only:
+                self.request = self.model.create_infer_request()
+            else:
+                logger.info(f"Compiling the Language model to {self._device} ...")
+                self.request = self._compile_model(
+                    self.model, self._device, self.ov_config, self.model_save_dir
+                ).create_infer_request()
         self._compile_text_emb()
 
     def _compile_text_emb(self):
         if self.text_emb_request is None:
             logger.info(f"Compiling the Text embeddings model to {self._device} ...")
-            self.text_emb_request = core.compile_model(self.text_emb_model, self._device, self.ov_config)
+            if self._compile_only:
```

Comment:
could you add a test for the `compile_only=True` case ? also we would need a test to make sure we don't break compatibility with a model that was already exported https://github.com/huggingface/optimum-intel/blob/v1.20.1/tests/openvino/test_modeling.py#L129

Reply:
we have no such models yet but ok I'll add

Reply:
added tests fr downloading from hub, compile_nly and also added tests for export